packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1521,12 +1521,21 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
     if (kFlutterMemoryAllocationsEnabled) {
       ChangeNotifier.maybeDispatchObjectCreation(this);
     }
-    if (kIsWeb || defaultTargetPlatform != TargetPlatform.android) {
+    final bool shouldListenToAppLifecycle = switch (defaultTargetPlatform) {
```

Comment:
Having the focus system automatically handle app lifecycle state changes is convenient in most cases, but what if a UI component wants to respond to app lifecycle events and user-initiated focus changes differently? If we do this automatically in the focus system, there's no way for a component to distinguish user-initiated focus changes and app-lifecycle focus changes.

For example a text field establishes / tears down an input connection based on whether it currently has focus.  Automatically unfocusing on app pause doesn't work well. It makes more sense to me that UI components should have logic to handle app lifecycle changes themselves (at least for text fields that's the case).

Reply:
Setting all UI components (not just text fields) to automatically unfocus/refocus based on app lifecycle state matches the behavior of other non-Flutter apps.

<details open> <summary>Example (Google Chrome)</summary>

![chrome focus changes](https://github.com/flutter/flutter/assets/10457200/bef42cd9-28e5-4214-b071-b7ef56b26609)

</details>

But the fact that a widget may want to differentiate between app lifecycle events and user-initiated focus changes is a great point.

In my mind, it makes sense to apply the iOS platform check today, and then sometime soon we should swap out the current app lifecycle listener for a more robust implementation that allows for the `TextField` input connection optimization you mentioned.